Orchestrator.py

```diff
@@ -128,29 +126,178 @@ def receive(self,frame):
+        dotBot['speed']   = speed
+
+    #=== Map
+
+    def _computeCellsExplored(self, ax, ay, bx, by):
+        '''
+        find cells passed through on trajectory between points a and b
+        example input - output :
+            {
+            'in': {
+                'ax': 1.10,
+                'ay': 1.20,
+                'bx': 2.20,
+                'by': 1.01,
+            },
+            'out': {
+                'cellsExplored': [(1.00,1.00), (1.50, 1.00), (2.00, 1.00)],
+                nextCell: (2.50, 1.00)
+                }
+        }
+        '''
+        returnVal = []
+
+        # set start and stop cells
+        (startX, startY) = (ax, ay)
+        (stopX, stopY)   = (bx, by)
+
+        # find current cell coordinates
+        (cx,cy)      = self._xy2cell(startX, startY)
+
+        # check if current cell is start cell
+        if startX == cx or startY == cy:
+            if (stopX < startX) and (startX == cx):
+                cx = cx - self.MINFEATURESIZE/2
+            if (stopY < startY) and (startY == cy):
+                cy = cy - self.MINFEATURESIZE/2
+
+        log.debug(f'moving from {startX}, {startY} to {stopX}, {stopY}')
+        log.debug(f'next cell -> {(cx, cy)}')
+
+        maxNumCells = ((abs(stopX - startX) + abs(stopY - startY))*4) + 2
+
+        # movement between cells (on boundaries)
+        if (
+            (startX == stopX) and (startX == cx or startX == cx+self.MINFEATURESIZE/2) or
+            (startY == stopY) and (startY == cy or startX == cy+self.MINFEATURESIZE/2)
+        ):
+            return {'cellsExplored': [], 'nextCell': None}
+
+        returnVal += [(cx, cy)]
+        exploredCellsComputed = False
+
+        if startX == stopX:
+            # vertical line, move up or down
+            while exploredCellsComputed == False:
+                xmax  = cx + self.MINFEATURESIZE/2
+                ymax  = cy + self.MINFEATURESIZE/2
+
+                if (
+                    cx <= stopX <= xmax and
+                    cy <= stopY <= ymax
+                ):
+                    exploredCellsComputed = True
+
+                if startY < stopY:
+                    cy = cy + self.MINFEATURESIZE/2
+                else:
+                    cy = cy - self.MINFEATURESIZE/2
+                returnVal += [(cx, cy)]
 
-        log.debug(f'dotbot {dotbot} is at ( {newX},{newY} ) ')
+                log.debug(f'num cells {len(returnVal)} and maxCellNum = {abs(maxNumCells)}')
+                assert len(returnVal) <= maxNumCells
+
+        else:
+
+            # move according to line equation y = mx + c
+            m  = (by - ay)/(bx - ax)
+            c  = startY - m*startX
+            log.debug(f'stop condition is {self._xy2cell(stopX, stopY)}')
+
+            while exploredCellsComputed == False:
+
+                ymin = cy
+                ymax = cy + self.MINFEATURESIZE/2
+                xmax = (cx + self.MINFEATURESIZE/2)
+
+                if stopX > startX:
```

Comment:
invert the you always use `<`, the smallest value on the left, the larger on the right